ui/src/api/api.ts

```diff
@@ -72,6 +88,175 @@ export async function getEntities(arxivId: string, getAllEntities?: boolean) {
+          ),
+          defining_formula_equations: (
+            response.sharedSymbolData[mathml]?.defining_formula_equations || []
+          ).map((s) => toLegacyRelationship(s, 'equation')),
+          definition_sentences: (
+            response.sharedSymbolData[mathml]?.definition_sentences || []
+          ).map((s) => toLegacyRelationship(s, 'sentence')),
+          snippet_sentences: (
+            response.sharedSymbolData[mathml]?.snippet_sentences || []
+          ).map((s) => toLegacyRelationship(s, 'sentence')),
+        },
+      };
+      return symbol;
+    } else if (isTerm(deduped)) {
+      const term: Term = {
+        id: deduped.id,
+        type: 'term',
+        attributes: {
+          bounding_boxes: deduped.attributes.bounding_boxes,
+          name: deduped.attributes.name,
+          source: 'tex-pipeline', // hardcoded since most everything comes out of the tex pipeline
+          tags: [],
+          term_type: deduped.attributes.term_type,
+          definitions: deduped.attributes.definitions,
+          definition_texs: deduped.attributes.definition_texs,
+          sources: deduped.attributes.sources,
+          snippets: deduped.attributes.snippets,
+        },
+        relationships: {
+          sentence: toLegacyRelationship(
+            deduped.relationships.sentence,
+            'sentence'
+          ),
+          definition_sentences: (
+            deduped.relationships.definition_sentences || []
+          ).map((s) => toLegacyRelationship(s, 'sentence')),
+          snippet_sentences: (
+            deduped.relationships.snippet_sentences || []
+          ).map((s) => toLegacyRelationship(s, 'sentence')),
+        },
+      };
+      return term;
+    }
+    // There's a new kind of entity that hasn't been implemented in the UI.
+    throw "Unknown entity type";
+  });
+  return {
+    data: entities,
+  };
+}
+
+
+/**
+ * This API returns a compacted representation of the paper's entities and their relationships.
+ * Certain fields of Symbol entity data are pulled into a separate `sharedSymbolData` map,
+ * which is organized by the Symbols' `attributes.disambiguated_id` value.
+ *
+ * NOTE: Currently, this function passes the response through a transform function to make
+ * it compatible with the existing UI code.
+ *
+ * @param arxivId arXiv ID of the viewed paper
+ * @param getAllEntities `true` retrieves entities of all types, `false` retrieves only citations
+ * @returns
+ */
+export async function getDedupedEntities(arxivId: string, getAllEntities?: boolean) {
+  const params = getAllEntities ? {
+    type: ENTITY_API_ALL
+  } : {};
+  const data = await doGet(
+    axios.get<EntityGetResponse>(
+      `/api/v0/papers/arxiv:${arxivId}/entities-deduped`,
+      {
+        params,
+        //@ts-ignore -- TODO: this pattern works in other projects, is there a version issue somewhere?
+        transformResponse: [].concat(axios.defaults.transformResponse).concat(undedupeResponse)
```

Comment:
The TS type checker wasn't happy with this line, complaining that the type for `axios.defaults.transformResponse` didn't match, which seems odd considering that's the default set of response transforms. I used axios on the Beaker project here and used this same pattern without type errors, so I'm wondering if maybe there's some combination of TS/axios versions that is causing the error.

Reply:
I think this is complaining because `[].concat()` takes an array as an argument. But you may as well do something like this:
```
transformResponse: [
  ...axios.defaults.transformResponse,
  undedupeResponse,
]
```